scio-avro/src/test/scala/com/spotify/scio/avro/AvroTapTest.scala

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2024 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.spotify.scio.avro
+
+import com.spotify.scio.ScioContext
+import com.spotify.scio.avro.AvroUtils._
+import com.spotify.scio.coders.Coder
+import com.spotify.scio.io.TapSpec
+import org.apache.avro.Schema
+import org.apache.avro.generic.{GenericRecord, GenericRecordBuilder}
+import com.spotify.scio.options.ScioOptions
+import com.spotify.scio.proto.SimpleV2.{SimplePB => SimplePBV2}
+import com.spotify.scio.proto.SimpleV3.{SimplePB => SimplePBV3}
+
+import java.nio.ByteBuffer
+
+class AvroTapTest extends TapSpec {
+  val schema: Schema = AvroUtils.schema
+  implicit def coder: Coder[GenericRecord] = avroGenericRecordCoder(schema)
+
+  "AvroTap" should "support saveAsAvroFile with SpecificRecord" in withTempDir { dir =>
```

Comment:
Wdyt about moving the `withTempDir` functionality into `scio-test-core`? It seems to have a lot of usage both within Scio and within many pipeline projects. (doesn't have to be in this PR!)

Reply:
Will expose this in a separate PR